src/main/java/com/lithium/dbi/rdbi/recipes/scheduler/DedupJobScheduler.java

```diff
@@ -0,0 +1,100 @@
+package com.lithium.dbi.rdbi.recipes.scheduler;
+
+import com.lithium.dbi.rdbi.Handle;
+import com.lithium.dbi.rdbi.RDBI;
+import org.joda.time.Instant;
+
+import java.util.List;
+
+/**
+ * Similar to ExclusiveJobScheduler. Main difference is that jobs are de-duplicates by state.
+ * So the same job might be in the ready queue and the running queue at the same time. However,
+ * only one instance of a given job will ever be found in a particular state. This allows clients
+ * to schedule a job to run immediately after the one currently in progress has been completed.
+ *
+ * When reserving multiple jobs the scheduler will make sure it returns the requested number of jobs as long
+ * as they are available to be reserved in the ready queue. This is an important difference from how the
+ * PriorityBasedJobScheduler and TimeBasedScheduler work.
+ */
+public class DedupJobScheduler extends AbstractDedupJobScheduler {
```

Comment:
ha i thought about that and figured you'd hate it. but now that you've suggested it consider it done